pkg/executor/update.go

```diff
@@ -351,7 +351,10 @@ func (*UpdateExec) handleErr(colName model.CIStr, rowIdx int, err error) error {
 	if types.ErrOverflow.Equal(err) {
 		return types.ErrWarnDataOutOfRange.GenWithStackByArgs(colName.O, rowIdx+1)
 	}
-
+	if types.ErrTruncatedWrongVal.Equal(err) && col != nil && col.ColumnInfo != nil && col.ColumnInfo.GetType() == mysql.TypeTimestamp {
+		ec := e.Ctx().GetSessionVars().StmtCtx.ErrCtx()
+		return errors.AddStack(ec.HandleErrorWithAlias(kv.ErrKeyExists, err, err))
```

Comment:
Related error handling logic in Insert statement: 
https://github.com/pingcap/tidb/blob/47179ae14d6bb7947d8857b4e8c91212bce999aa/pkg/executor/insert_common.go#L324-L328

I've added error type and column type checking to avoid mishandling other errors here.